rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -183,66 +175,186 @@ def _query_blockchain_info_has_transactions(
+            raw_tx_list=self._query_blockchain_info(accounts=accounts, key='txs'),
+            options=options,
+            deserialize_fn=self._deserialize_tx_from_blockchain_info,
+        )
+
+    def _deserialize_tx_from_blockcypher(
+            self,
+            data: dict[str, Any],
+    ) -> BitcoinTx:
+        """Deserialize a raw tx dict from blockcypher into a BitcoinTx.
+        If the tx has a large number of TxIOs, the remaining TxIOs will be queried using the urls
+        provided in the API response.
+        May raise DeserializationError, KeyError, ValueError.
+        """
+        inputs: list[dict[str, Any]] = []
+        outputs: list[dict[str, Any]] = []
+        for side, tx_io_list in (('inputs', inputs), ('outputs', outputs)):
+            next_data = data.copy()
+            while True:
+                tx_io_list.extend(list_chunk := next_data[side])
+                if (
+                    (next_url := next_data.get(f'next_{side}')) is None or
+                    len(list_chunk) < BLOCKCYPHER_TX_IO_LIMIT
+                ):
+                    break  # all TxIOs for this side have been queried
+
+                next_data = request_get_dict(url=next_url, handle_429=True, backoff_in_seconds=1)
+
+        return BitcoinTx(
+            tx_id=data['hash'],
+            timestamp=deserialize_timestamp_from_date(
+                date=data['confirmed'],
+                formatstr='iso8601',
+                location='blockcypher bitcoin tx',
+            ),
+            block_height=deserialize_int(data['block_height']),
+            fee=satoshis_to_btc(data['fees']),
+            inputs=BtcTxIO.deserialize_list(
+                data_list=inputs,
+                direction=BtcTxIODirection.INPUT,
+                deserialize_fn=BtcTxIO.deserialize_from_blockcypher,
+            ),
+            outputs=BtcTxIO.deserialize_list(
+                data_list=outputs,
+                direction=BtcTxIODirection.OUTPUT,
+                deserialize_fn=BtcTxIO.deserialize_from_blockcypher,
+            ),
+        )
+
+    def _query_blockcypher_transactions(
+            self,
+            accounts: Sequence[BTCAddress],
+            options: dict[str, Any],
+    ) -> tuple[int, list[BitcoinTx]]:
+        """Query blockcypher for transactions.
+        Txs from the api are ordered newest to oldest, with pagination via block_height.
+        Returns a tuple containing the latest queried block height and the list of txs.
+        """
+        accounts_tx_lists: dict[BTCAddress, list[dict[str, Any]]] = defaultdict(list)
+        for i in range(0, len(accounts), BLOCKCYPHER_BATCH_SIZE):
+            accounts_chunk = list(accounts[i:i + BLOCKCYPHER_BATCH_SIZE])
+            before_height = None
+            while len(accounts_chunk) > 0:
+                url = f"{BLOCKCYPHER_BASE_URL}/addrs/{';'.join(accounts_chunk)}/full?limit={BLOCKCYPHER_TX_LIMIT}&txlimit={BLOCKCYPHER_TX_IO_LIMIT}"  # noqa: E501
```

Comment:
you can do this instead.
```py
request_data = {
    'params': {
        'limit': BLOCKCYPHER_TX_LIMIT,
        'tx_limit': BLOCKCYPHER_TX_IO_LIMIT,
    }
}
while len(account_chunk) > 0:
    request_data['url'] = f'{BLOCKCYPHER_BASE_URL}/addrs/{';'.join(accounts_chunk)}/full'
    if before_height is not None:
        request_data['params']['before'] = before_height
 ```
 
 then you can pass an optional `params` argument to request_get function signature.